adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala

```diff
@@ -503,19 +605,53 @@ case class ReferenceRegion(
   }
 
   /**
+   * Determines if two regions are colocated on the same strand.
+   *
+   * @param other The other region.
+   * @return True if the two are on the same strand, false otherwise
+   */
+  def sameStrand(other: ReferenceRegion): Boolean = {
```

Comment:
JIT _could_, but you would have to make the method `final` (or `private`). Even then, I am not sure that JIT could inline the function since `other` is a `ReferenceRegion` which is not a `final class`.